SunGather/registers-sungrow.yaml

```diff
@@ -374,7 +374,7 @@ registers:
       level: 2    
       address: 5036
       datatype: "U16"
-      accuracy: 0.1
+      accuracy: 0.01
```

Comment:
On the other inverters (for me it's a SH10RT) the accuracy of 0.1 is correct.

Reply:
It depends on if modbus or http method… mine is 0.1 for http and 0.01 on modus. Annoying.

Reply:
For me it's 0.1 for modbus, even more annoying.

Reply:
Woosh - are you both running SH10RTs then ?

Reply:
The modbus protocol and registers are the same for PV, hybrid and battery. See the official documentation. So there “shouldn’t” be a difference in model. The issue is the register is returning different accuracy pending on how you access it, not different model.

Reply:
Ya, that makes sense, I was merely wanting to rule in/out different models in the mix between what you both were seeing.  Does it warrant some sort of "quirks" mechanism where we can augment the register data with alternative accuracies depending on model and/or access mode ?  And if this is a better thing to float on Discourse happy to do so and mebbe even take a run at implementing something.

Reply:
> The modbus protocol and registers are the same for PV, hybrid and battery. See the official documentation. So there “shouldn’t” be a difference in model. The issue is the register is returning different accuracy pending on how you access it, not different model.

Wouldn't that be considered a defect in the http wrapper to modbus? Or is it the inverter that actually use slightly different specs depending on whether you use modbus or http?

Reply:
I would consider this a bug, too.

Reply:
All it does is read the data the inverter returns. The bug is in the WiNet-S itself. 
Anyways, my current working version of SunGather has the “official” registers file and then a “overrides” file so people can make easier changes. This also means quirks like this can have an override option.

Reply:
FWIW I was able to confirm overnight that Register 13021 (Battery current) is a S16 rather than U16. It's a positive value when the battery is providing power out to the inverter, negative value when the battery is being charged (this is with a Sungrow SBR160 battery, but I would imagine it's independent of the battery type)